xhttp/client.go

```diff
@@ -158,7 +159,26 @@ func (r *retrierClient) do(ctx context.Context, req *http.Request, requestBody [
 		}
 		log.Debug("xhttp.Client: retrying request with error status code")
 		r.onRetry(req, res, err)
-		// Maybe add handling for Retry-After header, so far this seems to be enough
+
+		// handle Retry-After header
+		const minSleepPeriod = time.Second
+		retryAfter := res.Header.Get("Retry-After")
+		requestedDuration, requestedTime, err := ParseRetryAfter(retryAfter)
+		switch {
+		case err != nil:
+			log.Warn(fmt.Sprintf("xhttp.Client: %v", err))
+		case requestedDuration > minSleepPeriod:
```

Comment:
```suggestion
		case requestedDuration >= minSleepPeriod:
```

1 second is valid, in this case 1 second would be ignored. Overriding the sleep function could help testing this kind of sneaky logic if necessary.